broker.go

```diff
@@ -1013,6 +1027,16 @@ func (b *Broker) sendInternal(rb protocolBody, promise *responsePromise) error {
 		return ErrUnsupportedVersion
 	}
 
+	// validate the request is using a supported API version
+	if b.brokerAPIVersions != nil && b.brokerAPIVersions[rb.key()] != [2]int16{0, 0} {
```

Comment:
[unnecessary guards] The test for `b.brokerAPIVersions != nil` is not necessary. It is not an error to access a `nil` map, it simply functions the same as an empty map. It is only assigning into a map that does not work.

Thus, when `b.brokerAPIVersions == nil` then `b.brokerAPIVersions[rb.key()] == [2]int16{ 0, 0 }`.

Reply:
Updated